allow multiple SitemapUrlProvider hosts

toooni opened the pull request:
International Projects often do have a multiple domain strategy (.com, .ch, .de ..). For this case i "replaced" the `host` config with `hosts`.
To avoid BC break, I kept the `host` configuration but had to make it optional. In my opinion it's alright because the command does throw an exception that no provider is registered. If somebody doesn't want to use the SitemapUrlProvider, it's the smoother solution anyway because `enable: false` is enough.

Maybe there's a chance the host config will be removed in a upcoming major version?

kbond commented on DependencyInjection/Configuration.php:
```diff
@@ -20,7 +20,10 @@ public function getConfigTreeBuilder()
+                        ->arrayNode('hosts')
+                            ->prototype('scalar')->end()
+                        ->end()
+                        ->scalarNode('host')->end()
```
Can you just add `->info('Deprecated')`

kbond commented on DependencyInjection/ZenstruckCacheExtension.php:
```diff
@@ -30,7 +30,12 @@ protected function loadInternal(array $mergedConfig, ContainerBuilder $container
+                $mergedConfig['sitemap_provider']['hosts'] = array_merge($mergedConfig['sitemap_provider']['hosts'], array($mergedConfig['sitemap_provider']['host']));
+            }
+
+            $container->setParameter('zenstruck_cache.sitemap_provider.hosts', $mergedConfig['sitemap_provider']['hosts']);
```
I think we should throw an exception is no hosts are set.

kbond commented on UrlProvider/SitemapUrlProvider.php:
```diff
@@ -18,13 +18,13 @@ class SitemapUrlProvider implements UrlProvider
      * @param string $host
```
`array`

kbond commented on UrlProvider/SitemapUrlProvider.php:
```diff
@@ -18,13 +18,13 @@ class SitemapUrlProvider implements UrlProvider
      * @param Client $client
      */
-    public function __construct($host, Client $client)
+    public function __construct($hosts, Client $client)
```
can you typehint `array`

kbond commented on Tests/DependencyInjection/ZenstruckCacheExtensionTest.php:
```diff
@@ -36,19 +36,20 @@ public function testWithHost()
-     * @expectedException \Symfony\Component\Config\Definition\Exception\InvalidConfigurationException
-     */
-    public function testMissingHost()
+    public function testWithHosts()
```
Can you add a test for just adding `host` to ensure BC? Also, can you keep the exception test if `hosts` and `host` isn't defined?

kbond commented:
Looks good! Just a few small things and I will merge.

kbond commented:
actually... technically v2.0.0 is still in alpha so... you can remove the `host` option.

toooni commented:
thank you for the fast response and the nice bundle btw ;)

- i removed the `host` config now completely and added an entry in the changelog
- made sure that the new `hosts` config can not be empty
- wrote a test to test the empty `hosts` config

kbond commented:
Awesome!
